lib/ui/screens/overview_screen.dart

```diff
@@ -38,7 +38,16 @@ class OverviewScreen extends StatelessWidget {
       ),
       body: CustomScrollView(
         slivers: [
-          _buildInfoBanner(context, user!),
+          FutureBuilder<Widget>(
+            future: _buildInfoBanner(context, user!),
```

Comment:
I am not sure if this is a good idea, to use `FutureBuilder` with a `Widget` 🤔 

[Documentation](https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html) states that:

> If the [future](https://api.flutter.dev/flutter/widgets/FutureBuilder/future.html) is created at the same time as the [FutureBuilder](https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html), then every time the [FutureBuilder](https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html)'s parent is rebuilt, the asynchronous task will be restarted.
> A general guideline is to assume that every build method could get called every frame, and to treat omitted calls as an optimization.

Thus if I understand correctly the `_buildInfoBanner(context, user!)` will be restarted everytime 🤔 

I think the correct approach would be to create a new `StatefulWidget` for info banner and create the future (`HelperService.canDonate(context)`) once in `initState` and then use `FutureBuilder` to observe it's result. See `_BoxesScreenState` for example. 

The same applies to the `NewOfferFloatingButton`.